### PR TITLE
PD-520: Upgrade Workspace Dependencies 

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

- Upgrades Workspace dependencies
- One test currently failing in Stage.spec.tsx (not sure why, didn't change any relevant code) so have to poke around there but wanted to get this up in the meantime

🎟 _Jira ticket:_ [PD-520](https://jira.mongodb.org/browse/PD-520)

## 🛠 Types of changes

- [x] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### CODE REVIEW

File: packages/icon/src/Icon.spec.tsx [hunk 1]
```diff
@@ -56,6 +56,8 @@ describe('packages/Icon/glyphs/', () => {
         });
       }
 
+      // expect statement wrapped in function
+      // eslint-disable-next-line
```
Comment (hswolff) [inline packages/icon/src/Icon.spec.tsx:60]: Better practice to say which rule you're disabling:
```suggestion
      // eslint-disable-next-line rule-goes-here
```

So you don't disable all eslint rules, just the one.

Docs: https://eslint.org/docs/2.13.1/user-guide/configuring#disabling-rules-with-inline-comments

File: packages/mongo-nav/src/on-element-click-provider/OnElementClick.spec.tsx [hunk 2]
```diff
@@ -64,10 +64,14 @@ describe('packages/mongo-nav/on-element-click-provider', () => {
     let el: keyof typeof defaultElements;
 
     for (el in defaultElements) {
+      // expect wrapped inside a function
+      // eslint-disable-next-line
```
Comment (hswolff) [inline packages/mongo-nav/src/on-element-click-provider/OnElementClick.spec.tsx:68]: Ditto these as well.

File: packages/mongo-nav/src/project-nav/ProjectNav.spec.tsx [hunk 3]
```diff
@@ -28,7 +28,7 @@ const Products = {
   Charts: 'charts',
 };
 
-export type Products = typeof Products[keyof typeof Products];
```
Comment (hswolff) [inline packages/mongo-nav/src/project-nav/ProjectNav.spec.tsx:31]: ?

Comment (brookescarlett) [reply packages/mongo-nav/src/project-nav/ProjectNav.spec.tsx:31]: Can't export from a spec file, but I dont believe we were importing this type anywhere

File: packages/mongo-nav/src/on-element-click-provider/OnElementClick.spec.tsx (hunk 2 above)
Comment (brookescarlett) [reply packages/mongo-nav/src/on-element-click-provider/OnElementClick.spec.tsx:68]: 👍

**Pull request merged**